frontend/src/components/App/icons.test.ts

```diff
@@ -0,0 +1,61 @@
+import { error } from 'console';
+import fs from 'fs';
+import glob from 'glob';
+import mdiIcons from './icons';
+
+// the usedIcons array is used to check that all icons are used in the frontend
+// we take the keys from the mdiIcons and add the prefix to it as a string
+const usedIcons = Object.keys(mdiIcons.icons).map(icon => `${mdiIcons.prefix}:${icon}`);
+const usedAliases = Object.keys(mdiIcons.aliases).map(alias => `${mdiIcons.prefix}:${alias}`);
+
+const checkIcons = async () => {
+  const files = glob.sync('./**/*.tsx', { ignore: './node_modules/**' });
+
+  const unusedIcons = [];
+
+  for (const file of files) {
+    const content = await fs.readFileSync(file, 'utf8');
+    // const regex = /mdi:\w+/g;
+    // \w+ matches any word character (equal to [a-zA-Z0-9_])
+    // + Quantifier — Matches between one and unlimited times, as many times as possible, giving back as needed (greedy)
+    // g modifier: global. All matches (don't return after first match)
+    // this will only read for words, not for words with a dash in it
+
+    // this regex will match all mdi:icon-name
+    // const regex = /mdi:[\w-]+/g;
+    // mdi: matches the characters mdi: literally (case sensitive)
+    // Match a single character present in the list below [\w-]+
+    // + Quantifier — Matches between one and unlimited times, as many times as possible, giving back as needed (greedy)
+    // \w matches any word character (equal to [a-zA-Z0-9_])
+    // - matches the character - literally (case sensitive)
+    // g modifier: global. All matches (don't return after first match)
+    // const regex = /mdi:[\w-]+/g;
+
+    const regex = /mdi:[\w-]+/g;
+
+    let match: any;
+    while ((match = regex.exec(content)) !== null) {
+      if (!usedIcons.includes(match[0]) && !usedAliases.includes(match[0])) {
+        unusedIcons.push({
+          icon: match[0],
+          file,
+        });
+      }
+    }
+  }
+
+  for (let i = 0; i < unusedIcons.length; i++) {
+    error(
+      `Icon ${unusedIcons[i].icon} from file ${unusedIcons[i].file} is not cached for offline use `
```

Comment:
Nit: Showing the line number would also be cool `file-name:line-number`, but if it's not directly given by the regex.exec, let's not complicate the more logic just for that.

Reply:
Interesting! I will push the notes edits and will try this if possible I'll push it next!

Reply:
Tried to find an easy way to implement this friday but could but was not forward enough to put in